app/controllers/mixins/checked_id_mixin.rb

```diff
@@ -10,7 +10,32 @@ def checked_item_id(hash = params)
     # Common routine to find checked items on a page (checkbox ids are
     # "check_xxx" where xxx is the item id or index)
     def find_checked_items(prefix = nil)
-      if params[:miq_grid_checks].present?
+      has_no_check = true
+      nested_list_item_key = ""
+      # Check if nested list items are tagged.
+      if params["miq_grid_checks"].present?
+        nested_list_item = params["miq_grid_checks"].split(",")[0]
+
+        # Create param string
+        nested_list_item_key = "select-row-#{nested_list_item}"
+      end
+
+      params.each do |var, val|
+        if var.starts_with?("check_")
+          has_no_check = false
+        else
+          next
+        end
+      end
+
+      # If id is present use id, unless when nested list exists, example on summary pages
+      if params[:id].present? && params[nested_list_item_key] != "on" && has_no_check
+        if params[:id].kind_of?(Array)
+          params[:id]
+        else
+          [params[:id]]
+        end
```

Comment:
```suggestion
        Array.wrap(params[:id])
```

Array.wrap method wraps its argument in an array unless it is already an array.